script/main.js

```diff
@@ -139,8 +141,31 @@ class Gauge extends Component {
   }
 }
 
+class DeviceSelector extends Component {
+
+  onValueChange(e) {
+    this.props.app.setState({
+      'device': e.target.value,
```

Comment:
Seems it doesn't work because you didn't refactor `state` into `{metrics, device, ...}` like I did in #18. The rest of app is gonna assume `device` is a metric and fail on lookups.

Reply:
yeah.. i tried dealing with that ... see L247.  but i didn't like it.
but ultimately it's definitely much better to split apart the state.